src/adhocracy_core/adhocracy_core/__init__.py

```diff
@@ -39,6 +39,11 @@ def add_after_commit_hooks(request):
         clear_transaction_changelog_after_commit_hook
     current_transaction = transaction.get()
     registry = request.registry
+    # Order matters here
+    if 'adhocracy.varnish_url' in registry.settings:
+        from adhocracy_core.caching import purge_varnish_after_commit_hook
```

Comment:
If you do the  the registry  check here instead inside the commit hook, 
you need to look at two different packages to understand whats going on.

Referening to our styleguide we should prefere separating exception handling from actual programm code. 
But i guess single responsibility is more important.